Library/Homebrew/cmd/pull.rb

```diff
@@ -402,7 +404,168 @@ def subject_for_bump(formula, old, new)
     subject
   end
 
+  def maybe_pbcopy(text)
+    pbcopy(text) unless ARGV.include? "--no-pbcopy"
+  end
+
   def pbcopy(text)
     Utils.popen_write("pbcopy") { |io| io.write text }
   end
+
+  # Publishes the current bottle files for a given formula to Bintray
+  def publish_bottle_files_to_bintray(f, creds)
+    repo = Bintray.repository(f.tap)
+    package = Bintray.package(f.name)
+    info = FormulaInfoFromJson.lookup(f.name)
+    version = info.pkg_version
+    curl "-w", '\n', "--silent", "--fail",
+         "-u#{creds[:user]}:#{creds[:key]}", "-X", "POST",
+         "-H", "Content-Type: application/json",
+         "-d", '{"publish_wait_for_secs": 0}',
+         "https://api.bintray.com/content/homebrew/#{repo}/#{package}/#{version}/publish"
+  end
+
+  # The URL for a given bottle
+  # Opts:
+  #  bottle_tag  - tag indicating platform of bottle. Defaults to current
+  #                process's platform
+  def bottle_url(f, opts = {})
+    my_bottle_tag = opts[:bottle_tag] || bottle_tag
+    my_bottle_tag ||= bottle_tag
+    spec = f.stable
+    bottle_spec = spec.bottle_specification
+    filename = Bottle::Filename.create(f, my_bottle_tag, bottle_spec.revision)
+    "#{bottle_spec.root_url}/#{filename}"
+  end
+
+  def bottle_url_any_platform(f)
+    info = FormulaInfoFromJson.lookup(f.name)
+    bottle_info = info.bottle_info_any
+    return bottle_info.nil? ? nil : bottle_info.url
+  end
+
+  # Formula info drawn from an external "brew info --json" call
+  class FormulaInfoFromJson
+    # The whole info structure parsed from the JSON
+    attr_accessor :info
+
+    def initialize(info)
+      @info = info
+    end
+
+    def self.lookup(name)
+      json = Utils.popen_read(HOMEBREW_BREW_FILE, "info", "--json=v1", name)
+      Homebrew.force_utf8!(json)
+      opoo "Failed getting info for #{name}" unless $?.success?
```

Comment:
Nope, not `ofail`; that just marks the command as failed. Looks like this should be `raise`. Changing.